RECENT_CHANGES.md

```diff
@@ -0,0 +1,26 @@
+# Recent Changes to the PDC Data Viewer
+
+## June 2023
+
+- Add the ability to show applicant data from data platform providers,
+  starting with Candid and Charity Navigator
+- Highlight current proposal in sidebar
+- Allow text selection on Dashboard table
+
+## May 2023
+
+- Add search capability to Dashboard
```

Comment:
I know that we added search capability in two stages: an initial "basic" search (which I believe was actually running client-side, IIRC), and then later that was replaced/improved by putting a real search engine into the Data Viewer.

So I expected either to see two items (at different chronological locations) about search, OR if this is the only item about search then I expected it to say something like "Improved search capability (added phrase searches, stemming, etc)" or something like that.

Reply:
@kfogel Since we deployed both search versions in May, I wasn't quite sure how to handle this one. Is it weirder to have "Add search" + "Improve search" as two separate items in May, or to have "Add and improve search" as a single item? (Using condensed language here.)

We could arguably fudge this one a _little_ and put "Add search" in April, because that is when the PR was opened. It just wasn't closed/deployed until May 1.

Reply:
@reefdog Oh, I see.  Well, I'm not sure we want to organize by calendar months anyway -- like, there might be several months when nothing major happens.  Ah, but I see the problem: we're using continuous deployment, so it's not like there's a "release" event.

The real releases, in a sense, have been the occasional Friday live demonstrations to the Pilot Participants group (and later, our release-equivalent will be demos or announcements to that group or whatever group succeeds it).  So maybe those announcements/demos should be the milestones that the recent changes list is organized by?

Reply:
@kfogel Yeah, our continuous release makes this a tricky problem!

If we think Friday demonstrations are a good milestone, I can reconstruct around those. Then perhaps—as you suggested on the call—we could have a "Pending" section at the top where new items accumulate until they're demoed.

I'd like to hear more input on this strategy before implementing it, since it'll require another pass through the history to reconstruct. If we don't get any feedback here we'll discuss tomorrow.

Reply:
Suppose there is a new, proposed change that has yet to be deployed. It sits in a Pull Request during review. Part of the change is the update to `RECENT_CHANGES`. No assumption can be made about the date when that change will be visible to the user because it has yet to pass review. I'm not sure what to suggest as a solution. Some kind of date reference is important. Perhaps the date should only reference past months and the top group is merely "Recent"? That way the list at the top can be built up with no assumed date, and when weeks or months pass, the date can be inserted.

Even though the software is only useful to people who actually use the software, I think the "done" definition should be "when it is available in production" rather than "when users get notified" or "when it gets demonstrated." This idea is about a "done" definition so it's indirectly related to the RECENT_CHANGES doc. Suppose a scheduled demo gets skipped or two demos happen within two weeks. I think I would prefer a more even interval in this document if there is an interval chosen at all.

Perhaps the most recent change should have no date on it, while each and every previous change needs to have a date associated. So if I were adding "...ability to show applicant data..." it might look like this:
```
- Add the ability to show applicant data from data platform providers,
  starting with Candid and Charity Navigator
- 2023-06-23 Highlight current proposal in sidebar
- 2023-06-23 Allow text selection on Dashboard table
...
```

And now that I try to do it, I see it is pretty tedious. I wish I had a better suggestion.
...
```

Reply:
@reefdog Your solution sounds fine, including the Pending section.

@bickelj Eh, easy enough to deal with the problem of the floating change: just include it in the PR, and then adjust the location right before merging.

Reply:
I like the monthly approach! In general, I'd prefer to minimize the amount of friction involved in making changes, and the monthly cadence feels like a useful trade-off between sharing updates without too much rebasing / merge conflicts / date tracking.

> there might be several months when nothing major happens

That's true! But I think it affects each approach similarly. If anything, grouping by month smooths over some of the demo scheduling concerns.

> I think the "done" definition should be "when it is available in production"

I agree!

I'd also love to hear feedback from our intended audience on any of this. *Do* they need or want more than a monthly update? Are they even going to click on the new link? Can we start with the lower-friction approach first and then refine it when we get that feedback?

All that said, I'll defer to @kfogel and @jim-mcgowan on this.

Reply:
Let's please just do monthly then; I agree with your assertion @jasonaowen that it's a good trade-off point between time-based and demo-based tracking.

It's my fault for asking the question about the two stages of search implementation in the first place :-).  I should have left well enough alone and not asked a fairly unimportant question that interfered with actually shipping this new file.  My bad!  I'll try to consider more carefully next time whether my question is *really* contributing.

Reply:
@kfogel Conversation that reinforces a direction can still be useful, and in this case I think it was! I'm glad you raised it.

Reply:
Yes, +1 to what @reefdog just said - I think that was a good conversation to be had, @kfogel! No fault to assign.